commit 3160ec1b188c607aaf7e783814269cdc0a06350e (refs/changes/42/3742/3)
Author: Patrick Mooney <pmooney@pfmooney.com>
Date:   2018-03-30T18:56:55+00:00 (1 year, 6 months ago)
    
    OS-6856 viona can miss msix fast path
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Approved by: Robert Mustacchi <rm@joyent.com>

diff --git a/usr/src/cmd/bhyve/pci_virtio_viona.c b/usr/src/cmd/bhyve/pci_virtio_viona.c
index 1c4b739711..a1da39d13b 100644
--- a/usr/src/cmd/bhyve/pci_virtio_viona.c
+++ b/usr/src/cmd/bhyve/pci_virtio_viona.c
@@ -671,6 +671,7 @@ pci_viona_write(struct vmctx *ctx, int vcpu, struct pci_devinst *pi,
 		assert(size == 2);
 		assert(sc->vsc_curq != VIONA_CTLQ);
 		sc->vsc_msix_table_idx[sc->vsc_curq] = value;
+		pci_viona_ring_set_msix(pi, sc->vsc_curq);
 		break;
 	case VIONA_R_CFG0:
 	case VIONA_R_CFG1:
